dev/data/6-project.json

```diff
@@ -3,18 +3,33 @@
     "kind": "Project",
     "metadata": {
       "name": "perses"
+    },
+    "spec": {
+      "display": {
+        "name": "perses"
+      }
```

Comment:
Added `spec.display.name` since **gopkg.in/yaml.v3** and **encoding/json** marshalers handle nil structs differently. Without this initialization, gopkg.in/yaml.v3 would omit the entire 'spec' field while encoding/json keeps it as an empty spec object.
You can see this behavior here: https://go.dev/play/p/kv0gT5GZt6M

Without a spec object the UI fails to load. 
![image](https://github.com/user-attachments/assets/3aa16969-17d6-4c39-9ad4-f654eb72a82d)

Reply:
mmm it's weird because the UI is getting a JSON not a YAML format so spec should be initialised in the API response.

Reply:
The issue is happening much before actually. The issue is happening during the provisioning of file database during the application start up. 

So, when the `config.database.file.extension` is "yaml" extension, a project manifest in `local_db/projects` folder  looks like the below snippet. 
```yaml
kind: Project
metadata:
    name: perses
    createdAt: 2024-10-24T18:48:14.174051Z
    updatedAt: 2024-10-24T18:48:14.174051Z
    version: 0
```
However, when the `config.database.file.extension` is "json", a project manifest in `local_db/projects` looks like the below snippet. 

```json
{
  "kind": "Project",
  "metadata": {
    "name": "incredibleproject",
    "createdAt": "2024-10-24T18:49:56.921669Z",
    "updatedAt": "2024-10-24T18:49:56.921669Z",
    "version": 0
  },
  "spec": {}
}
```

The marshalling happens in [this line](https://github.com/perses/perses/blob/6232c108bf11bb3e6fef9d658e7b8c1a6fa1cacb/internal/api/database/file/file.go#L294)

Reply:
What I mean is it shouldn't happen actually.

Because when the frontend is requesting the backend for anything, the response will be marshalled in JSON. So whatever the format is in the database, once it is sent to the frontend, the format is JSON.

So probably there is another thing to fix there. But it's not that urgent.